proto/redpanda/api/dataplane/v1/quota.proto

```diff
@@ -0,0 +1,222 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+import "redpanda/api/auth/v1/authorization.proto";
+
+message Quota {
+  // The type of entity that the quota applies to.
+  enum EntityType {
+    ENTITY_TYPE_UNSPECIFIED = 0;
+    ENTITY_TYPE_ANY = 1;
+    ENTITY_TYPE_CLIENT_ID = 2;
+    ENTITY_TYPE_CLIENT_ID_PREFIX = 3;
+    ENTITY_TYPE_USER = 4; // Not supported in Redpanda
+    ENTITY_TYPE_IP = 5; // Not supported in Redpanda
+  }
+
+  // The type of value that the quota represents.
+  enum ValueType {
+    VALUE_TYPE_UNSPECIFIED = 0;
+    VALUE_TYPE_ANY = 1;
+    VALUE_TYPE_PRODUCER_BYTE_RATE = 2;
+    VALUE_TYPE_CONSUMER_BYTE_RATE = 3;
+  }
+
+  message Entity {
+    Quota.EntityType entity_type = 1;
+    string entity_name = 2;
+  }
+
+  message Value {
+    Quota.ValueType value_type = 1;
+    int64 value = 2;
+  }
+}
+
+message ListQuotasRequest {
+  // If the filter_type is FILTER_TYPE_NAME, then the name of the entity must be specified.
+  option (buf.validate.message).cel = {
+    id: "name_required_for_filter_type_name"
+    message: "name is required when filter_type is FILTER_TYPE_NAME"
+    expression: "this.filter_type != 2 || this.name != ''"
+  };
+  // FilterType defines the type of filter to apply when listing quotas.
+  enum FilterType {
+    FILTER_TYPE_UNSPECIFIED = 0;
+    // The filter type any will return quotas for all entities.
+    FILTER_TYPE_ANY = 1;
+    // The filter type name when specified with name
+    // will return quotas for the specified client id.
+    FILTER_TYPE_NAME = 2;
+    // The filter type default returns the default client id
+    FILTER_TYPE_DEFAULT = 3;
+  }
+  Quota.EntityType entity_type = 1;
+  FilterType filter_type = 2;
+  string name = 3;
+}
+
+message ListQuotasResponse {
+  // QuotaEntry represents the quota for a specific entity type.
+  message QuotaEntry {
+    Quota.Entity entity = 1;
+    repeated Quota.Value values = 2;
+  }
+  repeated QuotaEntry quotas = 1;
+}
+
+message RequestEntity {
```

Comment:
I think this name is too generic since it would be valid for the entire dataplane package.